honeybee_radiance/view.py

```diff
@@ -738,31 +738,90 @@ def to_file(self, folder, file_name=None, mkdir=False):
+
+        Args:
+            angle: An angle in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated. If None, self.position is used. (Default: None).
+        """
+        view_up_vector = pv.Vector3D(*self.up_vector)
+        view_position = pv.Point3D(*self.position)
+        view_direction = pv.Vector3D(*self.direction)
+        view_plane = Plane(n=view_up_vector, o=view_position, x=view_direction)
+        position = origin if origin is not None else view_position
+
+        rotated_plane = view_plane.rotate_xy(math.radians(angle), position)
+        self._apply_plane_properties(rotated_plane, view_direction, view_up_vector)
 
-    def rotate(self, angle, axis=None, position=None):
-        """Rotate view around an axis.
+    def reflect(self, plane):
+        """Reflect this view across a plane.
 
         Args:
-            angle: Rotation angle in radians.
-            axis: Rotation axis as a Vector3D (Default: self.up_vector).
-            position: Rotation position point as a Point3D (Default: self.position)
+            plane: A ladybug_geometry Plane across which the object will
+                be reflected.
         """
         view_up_vector = pv.Vector3D(*self.up_vector)
         view_position = pv.Point3D(*self.position)
         view_direction = pv.Vector3D(*self.direction)
-        view_plane = plane.Plane(n=view_up_vector, o=view_position, x=view_direction)
-        axis = pv.Vector3D(*axis) or view_up_vector
-        position = pv.Point3D(*position) or view_position
+        view_plane = Plane(n=view_up_vector, o=view_position, x=view_direction)
+
+        ref_plane = view_plane.reflect(plane.n, plane.o)
+        self._apply_plane_properties(ref_plane, view_direction, view_up_vector)
 
-        rotated_plane = view_plane.rotate(axis, angle, position)
+    def scale(self, factor, origin=None):
+        """Scale this view by a factor from an origin point.
 
-        self.position = rotated_plane.o
-        self.direction = rotated_plane.x
-        self.up_vector = rotated_plane.n
+        Args:
+            factor: A number representing how much the object should be scaled.
+            origin: A ladybug_geometry Point3D representing the origin from which
+                to scale. If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        view_position = pv.Point3D(*self.position)
+        self.position = view_position.scale(factor, origin)
+        self.direction = pv.Vector3D(*self.direction) * factor
+        self.up_vector = pv.Vector3D(*self.up_vector) * factor
+
+    def _apply_plane_properties(self, plane, view_direction, view_up_vector):
+        """Re-set the position, direction and up_vector from a Plane.
+
+        This method also ensures that the magnitude of the vectors is unchanged
+        (since all Plane objects will have unitized vectors).
```

Comment:
Hmmm! I have to check but I think in Radiance the vector length doesn't make a difference and we can keep it always unitized.

Reply:
If it didn't make a difference, then I imagine that you would not have the following docstring in the view class:
```
direction: Set the view direction (-vd) vector to (x, y, z). The
    length of this vector indicates the focal distance as needed by
    the pixel depth of field (-pd) in rpict. Default: (0, 0, 1)
```

Reply:
That's from Radiance's documentation and you are right.